devito/ir/clusters/cluster.py

```diff
@@ -332,6 +332,12 @@ def dspace(self):
         intervals = intervals.promote(lambda d: not d.is_Sub)
         intervals = intervals.zero(set(intervals.dimensions) - oobs)
 
+        # Buffered TimeDimensions should not shirnk their upper time offset
+        for f, v in parts.items():
```

Comment:
this is better than before, but it's breaching through abstractions pretty heavily...

as you probably know, you shouldn't use the "time" inside the compiler, for obvious reasons

Reply:
what's wrong with setting an interval to 0,0 if the dimension is something like `d.is_NonlinearDerived` OR `d.is_Stepping` or ... to be determined?

https://github.com/devitocodes/devito/pull/2237/files#diff-908f8895b205a02f261796e377b0b34ab9f3fe9998a5e152e2c8ec102e8ab3e7R305

Reply:
I have not looked into these properties TBH I am not sure if they apply to all cases
Defeinitly it is dangerous to zero the lower bound as well...
several checkpointing/gradient/buffering tests fail.
I will try to find some solution for this. Thanks for the ideas!